regolith/tools.py

```diff
@@ -2315,3 +2315,42 @@ def get_appointments(person, appointments, target_grant=None):
             appointments.append((person.get('_id'), bd, ed, appt.get('loading'),
                          round(weighted_duration, 2), appt.get('grant')))
     return appointments
+
+def add_todo_id_to_tasks(gtx, client, db, coll, milestone_id, task_id):
+    pdoc, upd_mil, all_miles, id, success = {}, [], [], [], []
+    target_mil = fragment_retrieval(gtx["projecta"], ["milestones"], milestone_id)
+    if target_mil:
+        for prum in target_mil:
```

Comment:
this looks rather inefficient.  Is there not a `get_one` that can pull this directly from the db?

Reply:
per convo with Tom, let's leave this as is.  `fragment_retrieval` is maybe not hte best name for that function though. 
I wonder whether we want a function called something like `get_subcollection(collection, key)` that would return a list if key in collection  is a list, else returns some kind of failure key.  This seems like a general thing we might want to do.  Then `get_subcollection(gtx["projecta"], "milestones")` would return the list of milestones and we can do things with the milestones (including search it for an id, for example `[milestone for milestone in milestones if milestone.get('milestone_id') == milestone_id]`

btw, I think gtx["projecta"] is a generator so it empties when you iterate through it.  We may want to turn it into a list before passing it to a function,  though it will work this way if we only use it once.

Reply:
back up a bit there, I am not sure what the functions are doing.  I guess the steps are:

1. figure out which prum contains the target milestone_id (this probably best done with an index when we figure out how to do that, but for now iterate through all the milestones in all the prums?). 
2. Get that prum
4. update tasks in that milestone by appending the task_uuid
5. update_one for that prum

It could then look like this.
```
target_prum = [prum for prum in prums if milestone.get("uuid") == target_uuid for milestone in prum.get('milestones') ]
if not target_prum:
    exception handling
else:
    for milestone in target_prum.get('milestones'):
        if milestone_id == target_id:
            milestone.get(tasks).append("uuid": task_id)
```
But I think if we had the get_subcollections it could look something like
```
for prum in prums:
    milestones = get_subcollection(prum, "milestones")
    milestone = [milestone for milestone in milestones if milestone.get("uuid") == target_id]
    tasks = get_subcollection(milestone[0], "tasks")
    tasks.append(task_id)
```
you will have to check if the append happens in place, in other words, does the whole prum object reflect this change (I am not sure), but if it does, then you would just have to do an `update_one` and push the updated prum to the db.

I am not sure about the syntax, but sthg like that.